opslevel/resource_opslevel_check_tool_usage.go

```diff
@@ -229,11 +229,11 @@ func (r *CheckToolUsageResource) ValidateConfig(ctx context.Context, req resourc
 		"tool_url_predicate":    configModel.ToolUrlPredicate,
 	}
 	for predicateSchemaName, predicate := range checkToolUsagePredicates {
-		predicateModel, diags := PredicateObjectToModel(ctx, predicate)
-		resp.Diagnostics.Append(diags...)
-		if predicateModel.Type.IsUnknown() || predicateModel.Type.IsNull() {
+		if predicate.IsNull() || predicate.IsUnknown() {
 			continue
 		}
+		predicateModel, diags := PredicateObjectToModel(ctx, predicate)
+		resp.Diagnostics.Append(diags...)
```

Comment:
~~Require an if statement for `resp.Diagnostics.HasError()` before this function ends~~ After reading more this is the way to do it so that each error in the loop is reported so the user can see all the attribute errors.